core/src/main/java/tc/oc/pgm/PGMPlugin.java

```diff
@@ -274,17 +274,40 @@ public void reloadConfig() {
     logger.setLevel(config.getLogLevel());
 
     for (String source : config.getMapSources()) {
-      final MapSourceFactory factory;
+      MapSourceFactory factory = null;
       try {
-        factory =
-            source.equalsIgnoreCase("default")
-                ? DefaultMapSourceFactory.INSTANCE
-                : new SystemMapSourceFactory(source);
+        if (source.equalsIgnoreCase("default"))
+          factory =
+              new GitMapSourceFactory(
+                  "https://github.com/KingOfSquares/DefaultMaps", logger, false);
+        if (source.startsWith("git;")) {
```

Comment:
Let's change the format to be the following:
```yml
map:
  folders:
    - "maps/"
  repositories:
    - uri: "https://github.com/Electroid/maps.git"
      branch: "tournament"
      folders:
        - "ctw/"
        - "dtc/"
```

You'll need to do some refactoring in `Config` and `PGMConfig`

Reply:
While i agree with this, special care should be taken not to over-complicate it for the user.
Just the URI should be enough to configure the git repo, all other fields should be optional for fine-tuning that the average user may not do

Reply:
I agree with Pablo. I can make it work with just a git link, but also work with branches/folders. Maybe also with .gitignore?

Reply:
Sure, you make make a single git link work as well.

Reply:
Your proposed config does not work with multiple repos. You can not have multiple ConfigurationSections with the same key. You would have to increment it with "repo", "repo1" "repo2" etc..

Reply:
You can, that's why the `-` is in front of the uri.

```yml
  repositories:
    - "https://github.com/Electroid/old-maps.git"
    - uri: "https://github.com/Electroid/other-maps.git"
    - uri: "https://github.com/Electroid/maps.git"
      branch: "tournament"
      folders:
        - "ctw/"
        - "dtc/"
```
Any of the following above should work.

Reply:
Whats the point of "folders"? You mentioned that it was not necessary to do the partial clone thing in this PR.

Reply:
If folders are specified, only those folders in the git repo are to be loaded. If folders is not specified, all folders in the git repo are loaded.